File: android/Reguerta/app/src/main/java/com/reguerta/user/domain/shifts/ShiftSwapRequest.kt
package com.reguerta.user.domain.shifts

enum class ShiftSwapRequestStatus {
    OPEN,
    CANCELLED,
    APPLIED,
}

enum class ShiftSwapResponseStatus {
    AVAILABLE,
    UNAVAILABLE,
}

data class ShiftSwapCandidate(
    val userId: String,
    val shiftId: String,
)

data class ShiftSwapResponse(
    val userId: String,
    val shiftId: String,
    val status: ShiftSwapResponseStatus,
    val respondedAtMillis: Long,
)

data class ShiftSwapRequest(
    val id: String,
    val requestedShiftId: String,
    val requesterUserId: String,
    val reason: String,
    val status: ShiftSwapRequestStatus,
    val candidates: List<ShiftSwapCandidate>,
    val responses: List<ShiftSwapResponse>,
    val selectedCandidateUserId: String?,
    val selectedCandidateShiftId: String?,
    val requestedAtMillis: Long,
    val confirmedAtMillis: Long?,
    val appliedAtMillis: Long?,
)

File: ios/Reguerta/Reguerta/Domain/Shifts/ShiftSwapRequest.swift
import Foundation

enum ShiftSwapRequestStatus: String, Equatable, Sendable {
    case open
    case cancelled
    case applied
}

enum ShiftSwapResponseStatus: String, Equatable, Sendable {
    case available
    case unavailable
}

struct ShiftSwapCandidate: Equatable, Sendable {
    let userId: String
    let shiftId: String
}

struct ShiftSwapResponse: Equatable, Sendable {
    let userId: String
    let shiftId: String
    let status: ShiftSwapResponseStatus
    let respondedAtMillis: Int64
}

struct ShiftSwapRequest: Identifiable, Equatable, Sendable {
    let id: String
    let requestedShiftId: String
    let requesterUserId: String
    let reason: String
    let status: ShiftSwapRequestStatus
    let candidates: [ShiftSwapCandidate]
    let responses: [ShiftSwapResponse]
    let selectedCandidateUserId: String?
    let selectedCandidateShiftId: String?
    let requestedAtMillis: Int64
    let confirmedAtMillis: Int64?
    let appliedAtMillis: Int64?
}
